### PR TITLE
Update to swift 3.0.1 and Kitura 1.1.0

### PR DESCRIPTION
**Pull request by rob-deans**
Related: #1

### CODE REVIEW

File: Sources/Application.swift [hunk 1]
```diff
@@ -64,13 +64,6 @@ public class Application {
 
         router.all("/api/*", middleware: BodyParser())
 
-        // Initialise Store
-        Model.store = MemoryStore() /*CloudantStore(ConnectionProperties(
```
Comment (tunniclm) [inline Sources/Application.swift:68]: Did you mean to remove this line in this pull request?

File: Sources/Application.swift [hunk 2]
```diff
@@ -30,7 +30,7 @@ public class Application {
 
         let fileURL = URL(fileURLWithPath: initialSearchPath)
         let directoryURL = fileURL.deletingLastPathComponent()
-
+
```
Comment (tunniclm) [inline Sources/Application.swift]: I don't think we want trailing whitespace here.

File: Sources/Application.swift [hunk 3]
```diff
@@ -160,10 +153,10 @@ public class Application {
...
             router.post(allPath) { req, res, next in
                 guard let contentType = req.headers["Content-Type"],
-                      contentType.hasPrefix("application/json") else {
-                    res.status(.unsupportedMediaType)
-                    res.send(json: JSON([ "error": "Request Content-Type must be application/json" ]))
-                    return next()
+                    contentType.hasPrefix("application/json") else {
+                        res.status(.unsupportedMediaType)
+                        res.send(json: JSON([ "error": "Request Content-Type must be application/json" ]))
+                        return next()
```
Comment (tunniclm) [inline Sources/Application.swift:161]: This indentation does not look good to me.

File: Sources/Application.swift [hunk 4]
```diff
@@ -203,10 +196,10 @@ public class Application {
...
             router.put(onePath) { req, res, next in
                 guard let contentType = req.headers["Content-Type"],
-                      contentType.hasPrefix("application/json") else {
-                    res.status(.unsupportedMediaType)
-                    res.send(json: JSON([ "error": "Request Content-Type must be application/json" ]))
-                    return next()
+                    contentType.hasPrefix("application/json") else {
+                        res.status(.unsupportedMediaType)
+                        res.send(json: JSON([ "error": "Request Content-Type must be application/json" ]))
+                        return next()
```
Comment (tunniclm) [inline Sources/Application.swift:204]: This indentation does not look good to me.

File: Sources/MemoryStore.swift [hunk 5]
```diff
@@ -61,17 +59,22 @@ class MemoryStore: Store {
...
-    func findAll(type: Model.Type, callback: @escaping EntitiesCallback) {
+    var entities: [[String:Any]] = []
+    var nextId: Int = 1
+
+    public init() {}
+
+    public func findAll(type: Model.Type, callback: @escaping EntitiesCallback) {
         let matchingEntities = entities.filter { $0["_type"] as! Model.Type == type }
-                                       .map(MemoryStore.sanitize)
+            .map(MemoryStore.sanitize)
```
Comment (tunniclm) [inline Sources/MemoryStore.swift:69]: Is there a good reason to change the indentation here, I thought it looked cleared when the functions were aligned.

File: Sources/MemoryStore.swift [hunk 6]
```diff
@@ -120,22 +123,22 @@ class MemoryStore: Store {
         }
     }
 
-    func update(type: Model.Type, id: ModelID, entity: [String:Any], callback: @escaping EntityCallback) throws {
+    public func update(type: Model.Type, id: ModelID, entity: [String:Any], callback: @escaping EntityCallback) throws {
         guard let memoryStoreID = id as? MemoryStoreID else {
             throw StoreError.idInvalid(id)
         }
-
+
```
Comment (tunniclm) [inline Sources/MemoryStore.swift]: I don't think we want trailing whitespace here.

File: Sources/MemoryStore.swift [hunk 7]
```diff
@@ -120,22 +123,22 @@ class MemoryStore: Store {
...
         }
-
+        
         guard let (existing, index) = findOne_(type: type, id: memoryStoreID) else {
             // NOTE(tunniclm): Only allowed to update existing models
             callback(nil, .notFound(memoryStoreID))
             return
         }
-
+
```
Comment (tunniclm) [inline Sources/MemoryStore.swift]: I don't think we want trailing whitespace here.

File: Sources/MemoryStore.swift [hunk 8]
```diff
@@ -120,22 +123,22 @@ class MemoryStore: Store {
...
             return
         }
-
+        
         var updatedItem = existing
         MemoryStore.mergeDictionary(&updatedItem, merge: entity)
         updatedItem["_type"] = type
         updatedItem["id"] = id
-
+
```
Comment (tunniclm) [inline Sources/MemoryStore.swift:141]: I don't think we want trailing whitespace here.

File: Sources/MemoryStore.swift [hunk 9]
```diff
@@ -151,12 +154,12 @@ class MemoryStore: Store {
                 updatedItem["id"] = newId
             }
         }
-
+
```
Comment (tunniclm) [inline Sources/MemoryStore.swift]: I don't think we want trailing whitespace here.

File: Sources/Model.swift [hunk 10]
```diff
@@ -92,7 +92,7 @@ open class Model {
 
     static func loadModels(fromDir url: URL) throws -> [(String, String)] {
         var failures: [(String, String)] = []
-
+
```
Comment (tunniclm) [inline Sources/Model.swift:95]: I don't think we want trailing whitespace here.

File: Sources/Model.swift [hunk 11]
```diff
@@ -229,7 +229,7 @@ open class Model {
                 throw ModelError.extraneousProperty(name: jsonPropertyName)
             }
             print("Found property definition for \(jsonPropertyName): \(property)") // DEBUG
-
+
```
Comment (tunniclm) [inline Sources/Model.swift:232]: I don't think we want trailing whitespace here.

File: Sources/Model.swift [hunk 12]
```diff
@@ -122,7 +122,7 @@ open class Model {
         // from ModelID to appropriate value to match model definition
         return try self.init(modelDict)
     }
-
+
```
Comment (tunniclm) [inline Sources/Model.swift]: I don't think we want trailing whitespace here.

**Review by tunniclm — COMMENTED** [review]
Sorry for all the nit-picks! The main issue is the removal of the `Model.store` assignment, which seems to be unintended here--shouldn't it be done in a separate feature that aligns with the `readConfig()` changes?

File: Sources/MemoryStore.swift (hunk 5 above)
Comment (rob-deans) [reply Sources/MemoryStore.swift:69]: This was done due to the auto-indent, can change that back?

**Comment by tunniclm** [comment]
@rob-deans This pull request has got a bit tangled up with multiple other features and contributions. I'm going to try and unpick it onto a new branch for you to review. Sorry about that!

**Comment by tunniclm** [comment]
@rob-deans The branch is `update_kitura_to_1_1` and can be viewed here: https://github.com/IBM-Swift/GeneratedSwiftServer/compare/update_kitura_to_1_1

Can you take a look and try it out? Let me know if I have made any errors or missed anything out. Thanks.

**Pull request closed without merging**